core/tests/test_models.py

```diff
@@ -0,0 +1,198 @@
+import secrets
+import string
+from collections.abc import Generator
+
+import pytest
+from django.contrib.auth import get_user_model
+
+from core.models import Community, Post, PostVote, Tag
+
+User = get_user_model()
+
+UPVOTE_SCORE = 1
+DOWNVOTE_SCORE = -1
+MULTIPLE_VOTES_SCORE = 2
+SCORE_3 = 3
+MIXED_VOTES_SCORE = 0
+
+
+def generate_random_password(length: int = 12) -> str:
+    alphabet = string.ascii_letters + string.digits + string.punctuation
+
+    return "".join(secrets.choice(alphabet) for i in range(length))
+
+
+@pytest.fixture()
+def user() -> Generator[User, None, None]:
+    return User.objects.create_user(
+        email="testuser@test.pl", password=generate_random_password(),
+    )
+
+
+@pytest.fixture()
+def community(user: User) -> Generator[Community, None, None]:
+    return Community.objects.create(user=user, name="Test Community")
+
+
+@pytest.fixture()
+def post(user: User, community: Community) -> Generator[Post, None, None]:
+    return Post.objects.create(
+        user=user,
+        community=community,
+        title="Test Post",
+        content="This is a test post",
+    )
+
+
+@pytest.mark.django_db()
+def test_post_score_initial(post: Post) -> None:
+    assert post.score == MIXED_VOTES_SCORE
+
+
+@pytest.mark.django_db()
+def test_post_score_upvote(post: Post, user: User) -> None:
+    vote = PostVote.objects.create(type=PostVote.UPVOTE)

```

Comment:
Would be good to have a method on Post like "vote", so we don't have to repeat this code.